src/material/core/tokens/_token-utils.scss

```diff
@@ -31,77 +33,97 @@ $_component-prefix: null;
+  $var-name: _create-var-name($_component-prefix, $token);
+  $fallback: _get-token-fallback($token, $fallback);
+  @return _create-var($var-name, $fallback);
+}
+
+// Assertion mixin that throws an error if the global state has not been set up by wrapping
+// calls with `use-tokens`.
+@function _assert-use-tokens($token) {
   @if $_component-prefix == null or $_tokens == null {
-    @error '`create-token-slot` must be used within `use-tokens`';
+    @error 'Function was not called within a wrapping call of `use-tokens`';
   }
   @if not map.has-key($_tokens, $token) {
     @error 'Token #{$token} does not exist. Configured tokens are: #{map.keys($_tokens)}';
   }
-  @if map.get($_tokens, $token) != null {
-    $fallback: null;
 
-    @if ($emit-fallback == true) {
-      $fallback: map.get($_tokens, $token);
-    }
-    @else if ($emit-fallback) {
-      $fallback: $emit-fallback;
-    }
+  @return true;
+}
 
-    $var-name: _get-css-variable($_component-prefix, $token);
-    $var-reference: if($fallback == null, var(#{$var-name}), var(#{$var-name}, #{$fallback}));
-    #{$property}: #{$var-reference};
+// Emits a slot for the given token, provided that it has a non-null value in the token map passed
+// to `use-tokens`.
+// Accepts an optional fallback parameter to include in the CSS variable.
+// If $fallback is `true`, then use the tokens map to get the fallback.
+@mixin create-token-slot($property, $token, $fallback: null) {
+  $_assert: _assert-use-tokens($token);
+  @if map.get($_tokens, $token) != null {
+    #{$property}: #{_get-token-value($token, $fallback)};
   }
 }
 
 // Returns the name of a token including the current prefix. Intended to be used in calculations
 // involving tokens. `create-token-slot` should be used when outputting tokens.
 @function get-token-variable-name($token) {
-  @if $_component-prefix == null or $_tokens == null {
-    @error '`get-token-variable` must be used within `use-tokens`';
-  }
-  @if not map.has-key($_tokens, $token) {
-    @error 'Token #{$token} does not exist. Configured tokens are: #{map.keys($_tokens)}';
-  }
-
-  @return _get-css-variable($_component-prefix, $token);
+  $_assert: _assert-use-tokens($token);
+  @return _create-var-name($_component-prefix, $token);
 }
 
 // Returns a `var()` reference to a specific token. Intended for declarations
 // where the token has to be referenced as a part of a larger expression.
-@function get-token-variable($token, $use-tokens-fallback: false, $fallback: null) {
-  @if $_component-prefix == null or $_tokens == null {
-    @error '`get-token-variable-reference` must be used within `use-tokens`';
-  }
-  @if not map.has-key($_tokens, $token) {
-    @error 'Token #{$token} does not exist. Configured tokens are: #{map.keys($_tokens)}';
-  }
-
-  $var: get-token-variable-name($token);
+// Accepts an optional fallback parameter to include in the CSS variable.
+// If $fallback is `true`, then use the tokens map to get the fallback.
```

Comment:
Is $fallback a boolean or a value? Based on this comment I'd assume it is a boolean, but looking at _get-token-fallback, $fallback appears to be a value

Reply:
This is really confusing =( it supports the arg `true` to mean "get the value from the current global tokens map", or otherwise it means "use this value exactly"

Ideally these fallbacks aren't set at all, since users should be calling our mixins to set the tokens properly. However this causes some client screenshot failures that we'll need to get through

Reply:
If not, then maybe revising the comment to more explicitly express this behavior

Reply:
I added a TODO for us to get rid of this use case for `true` altogether. We just need to fix failing client screenshots for this case. This function is overly complex and I want to fix it, but I'd prefer to do it in a followup